depthai_nodes/ml/parsers/base_parser.py

```diff
@@ -0,0 +1,33 @@
+from abc import ABCMeta, abstractmethod
+from typing import Dict, Union
+
+import depthai as dai
+
+
+class BaseMeta(ABCMeta, type(dai.node.ThreadedHostNode)):
+    pass
+
+
+class BaseParser(dai.node.ThreadedHostNode, metaclass=BaseMeta):
+    """Base parser class for neural network output parsers."""
+
+    def __init__(self):
+        super().__init__()
+
+    @property
+    @abstractmethod
+    def input(self) -> dai.Node.Input:
+        pass
+
+    @property
+    @abstractmethod
+    def output(self) -> dai.Node.Output:
+        pass
+
+    @abstractmethod
+    def build(self, head: Union[dai.NNArchive, Dict]):
```

Comment:
Moreover, if I get it correctly, one can now initialize a parser from a `dai.NNArchive` object only through the `Parser` class which decodes heads and constructs parsers from head configs. So is the type hint `dai.NNArchive` even necessary here?

Reply:
The Parser has to return a dict of values so that dai team can link it up correctly. If dict has only one key-value pair, it will return ouput as expected. 

Yes you can only initialize it through the parser, this was done in an effort to enforce its usage and to make nn_archive the ground truth / center for all of our apps. Yes sorry I missed a type hint, will fix in next commit.